### PR TITLE
new Events and localization for AttachResult and ResultsFilter components

### PR DESCRIPTION
**Pull request by mikegron**
Advise if I should add all events/strings in a single file for every component instead ?

Added strings for defaults strings of AttachResult and ResultsFilter
Added events when attaching / detaching result and filtering
SNOW-182

### CODE REVIEW

**Comment by louis-bompart** [comment]
Take the habit of creating branches from your fork :P

File: tests/components/ResultsFilter/ResultsFilter.spec.ts [hunk 1]
```diff
@@ -42,6 +43,23 @@ describe('ResultsFilter', () => {
         expect(filter.cmp.isSelected()).toBeFalsy();
     });
 
+    it('should trigger events on toggling', done => {
+        let cpt = 0;
+        Coveo.$$(filter.env.root).on(ResultsFilterEvents.Click, (evt: Event, args: IResultsFilterEventArgs) => {
+            expect(evt.type).toBe(ResultsFilterEvents.Click);
+            if (args.checked) {
```
Comment (louis-bompart) [inline tests/components/ResultsFilter/ResultsFilter.spec.ts]: What's the point of this if-else ?

File: tests/components/ResultsFilter/ResultsFilter.spec.ts [hunk 2]
```diff
@@ -42,6 +43,23 @@ describe('ResultsFilter', () => {
...
+                cpt++;
+            } else {
+                cpt++;
+            }
+            if (cpt == 2) {
+                done();
+            }
+        });
+        filter.cmp.toggle();
```
Comment (louis-bompart) [inline tests/components/ResultsFilter/ResultsFilter.spec.ts]: Double toggle is normal?

File: src/components/ResultsFilter/Strings.ts [hunk 3]
```diff
@@ -0,0 +1,5 @@
+import { Translation, Language } from '../../utils/translation';
```
Comment (louis-bompart) [inline src/components/ResultsFilter/Strings.ts]: Need to add this file to the side-effects of package.json

File: src/components/AttachResult/Strings.ts [hunk 4]
```diff
@@ -0,0 +1,6 @@
+import { Translation, Language } from '../../utils/translation';
```
Comment (louis-bompart) [inline src/components/AttachResult/Strings.ts]: Need to add this file to the side-effects of package.json

File: tests/components/ResultsFilter/ResultsFilter.spec.ts (hunk 1 above)
Comment (mikegron) [reply tests/components/ResultsFilter/ResultsFilter.spec.ts]: Make sure when I toggle two times the event has both values

File: tests/components/ResultsFilter/ResultsFilter.spec.ts (hunk 2 above)
Comment (mikegron) [reply tests/components/ResultsFilter/ResultsFilter.spec.ts]: yeah alltough I could split this test in two maybe

File: src/components/AttachResult/Strings.ts (hunk 4 above)
Comment (mikegron) [reply src/components/AttachResult/Strings.ts]: what does it do exactly ? 🤔

Comment (jeremierobert-coveo) [reply src/components/AttachResult/Strings.ts]: That's false

File: tests/components/ResultsFilter/ResultsFilter.spec.ts (hunk 1 above)
Comment (louis-bompart) [reply tests/components/ResultsFilter/ResultsFilter.spec.ts]: Yeah but... You have exactly the same statements in the if and else

File: tests/components/ResultsFilter/ResultsFilter.spec.ts [hunk 5]
```diff
@@ -42,6 +43,23 @@ describe('ResultsFilter', () => {
         expect(filter.cmp.isSelected()).toBeFalsy();
     });
 
+    it('should trigger events on toggling', done => {
+        let cpt = 0;
+        Coveo.$$(filter.env.root).on(ResultsFilterEvents.Click, (evt: Event, args: IResultsFilterEventArgs) => {
```
Comment (jeremierobert-coveo) [inline tests/components/ResultsFilter/ResultsFilter.spec.ts]: I think you can simplify this by giving a spy as callback ?

File: src/components/AttachResult/Strings.ts (hunk 4 above)
Comment (louis-bompart) [reply src/components/AttachResult/Strings.ts]: To be fairly honest, I'm quite new to the concept of tree shaking and side-effects, so I won't be the best to explains, but I can give you a link to the documentation: https://webpack.js.org/guides/tree-shaking/ 😅

File: tests/components/ResultsFilter/ResultsFilter.spec.ts [hunk 6]
```diff
@@ -42,6 +44,33 @@ describe('ResultsFilter', () => {
...
+            });
+            filter.cmp.toggle();
+        });
+    });
+
+    it('should trigger event with checked false when toggling twice', () => {
+        return new Promise(resolve => {
+            filter.cmp.toggle();
+            Coveo.$$(filter.env.root).on(ResultsFilterEvents.Click, (evt: Event, args: IResultsFilterEventArgs) => {
+                expect(evt.type).toBe(ResultsFilterEvents.Click);
```
Comment (louis-bompart) [inline tests/components/ResultsFilter/ResultsFilter.spec.ts]: Not sure if there's real value on this expect.

File: tests/components/ResultsFilter/ResultsFilter.spec.ts [hunk 7]
```diff
@@ -42,6 +44,33 @@ describe('ResultsFilter', () => {
         expect(filter.cmp.isSelected()).toBeFalsy();
     });
 
+    it('should trigger event with checked true when first toggling', () => {
+        return new Promise(resolve => {
+            Coveo.$$(filter.env.root).on(ResultsFilterEvents.Click, (evt: Event, args: IResultsFilterEventArgs) => {
+                expect(evt.type).toBe(ResultsFilterEvents.Click);
```
Comment (louis-bompart) [inline tests/components/ResultsFilter/ResultsFilter.spec.ts]: Not sure if there's real value on this expect.

File: src/components/AttachResult/AttachResult.ts [hunk 8]
```diff
@@ -37,12 +40,12 @@ export class AttachResult extends Component {
     private buttonElement: HTMLElement;
     private tooltipElement: HTMLElement;
 
-    static options: IAttachResultOptions = {
+    public static readonly options: IAttachResultOptions = {
```
Comment (louis-bompart) [inline src/components/AttachResult/AttachResult.ts]: Add a JSDOC comment here

**Pull request merged**